examples/6. Symbolics/6. Symbolics.jl

```diff
@@ -64,26 +72,29 @@ x = MechanismState(double_pendulum);
 
 # Set the joint configuration vector of the MechanismState to a new vector of symbolic variables
 q = configuration(x)
-for i in eachindex(q)
-    q[i] = symbols("q_$i", real=true)
-end
+q .= Symbolics.variables(:q, 1:length(q))
 
 # Set the joint velocity vector of the MechanismState to a new vector of symbolic variables
 v = velocity(x)
-for i in eachindex(v)
-    v[i] = symbols("v_$i", real=true)
-end
+v .= Symbolics.variables(:v, 1:length(q))
 
 
 # ## Compute dynamical quantities in symbolic form
 
 # Mass matrix
-simplify.(mass_matrix(x))
+simplify.(mass_matrix(x), expand=true)
 
 
 # Kinetic energy
-simplify(kinetic_energy(x))
+simplify(kinetic_energy(x), expand=true)
 
 
 # Potential energy
-simplify(gravitational_potential_energy(x))
+try
+    ## This throws `ERROR: TypeError: non-boolean (Num) used in boolean context` because
+    ## of `m > 0 || return zero(cache_eltype(state))` in `gravitational_potential_energy`.
+    ## See https://docs.sciml.ai/Symbolics/stable/manual/faq/ for more details.
+    simplify(gravitational_potential_energy(x), expand=true)
+catch e
+    e
+end
```

Comment:
This cell is not working because of the short-circuiting on this line:
https://github.com/JuliaRobotics/RigidBodyDynamics.jl/blob/d6df4ebeb927e98bfa5216997c5d08502273189c/src/mechanism_state.jl#L900

I tried removing the short-circuiting statement and it seems to fix it, but the repercussions of doing that are not clear to me. I am afraid that not having the early return might affect performance somehow, but on the other hand I could not see it being used as part of other methods...

@tkoolen would be the best person to chip in. 😛